datamodel.py

```diff
@@ -0,0 +1,164 @@
+"""Implementation of the Datamodel for the parts list with natural sort."""
+
+import logging
+import re
+
+import wx.dataview as dv
+
+from .helpers import loadIconScaled
+
+BOM_COL = 6
+POS_COL = 7
+SIDE_COL = 9
+
+
+class PartListDataModel(dv.PyDataViewModel):
+    """Datamodel for use with the DataViewCtrl of the mainwindow."""
+
+    def __init__(self, scale_factor):
+        super().__init__()
+        self.data = []
+        self.bom_pos_icons = [
+            loadIconScaled(
+                "mdi-check-color.png",
+                scale_factor,
+            ),
+            loadIconScaled(
+                "mdi-close-color.png",
+                scale_factor,
+            ),
+        ]
+        self.side_icons = [
+            loadIconScaled(
+                "TOP.png",
+                scale_factor,
+            ),
+            loadIconScaled(
+                "BOT.png",
+                scale_factor,
+            ),
+        ]
+        self.logger = logging.getLogger(__name__)
+
+    @staticmethod
+    def natural_sort_key(s):
+        """Return a tuple that can be used for natural sorting."""
+        return [
+            int(text) if text.isdigit() else text.lower()
+            for text in re.split("([0-9]+)", s)
+        ]
+
+    def GetColumnCount(self):
+        """Get number of columns."""
+        return 10
```

Comment:
can we return the columns used in GetColumnType() and use len() on that to avoid hardcoding this value?

Reply:
Done!